src/CaptureClient/CaptureEventProcessor.cpp

```diff
@@ -550,12 +552,33 @@ void CaptureEventProcessorForListener::ProcessThreadNamesSnapshot(
   }
 }
 
+[[nodiscard]] static ThreadStateSliceInfo::WakeupReason toWakeupReason(
+    orbit_grpc_protos::ThreadStateSlice::WakeupReason reason) {
+  switch (reason) {
+    case orbit_grpc_protos::ThreadStateSlice::WakeupReason::
+        ThreadStateSlice_WakeupReason_kNotApplicable:
+      return ThreadStateSliceInfo::WakeupReason::kNotApplicable;
+    case orbit_grpc_protos::ThreadStateSlice::WakeupReason::
+        ThreadStateSlice_WakeupReason_kUnblocked:
+      return ThreadStateSliceInfo::WakeupReason::kUnblocked;
+    case orbit_grpc_protos::ThreadStateSlice::WakeupReason::ThreadStateSlice_WakeupReason_kCreated:
+      return ThreadStateSliceInfo::WakeupReason::kCreated;
+    default:
```

Comment:
Is it possible to remove the `default` and add the remaining cases (it should be some internal value)? This way, if an enum value is ever added, we get notified by a compilation error.

Reply:
I did remove the default but there are a couple of values from the proto enum called INT_MIN_SENTINAL and INT_MAX_SENTINAL that I had to populate. I made those values give an ORBIT_UNREACHABLE.

Reply:
Okay, so I removed the default statement however, gcc10 and clang7 compilers complained that there's no return statement in a void function after the switch. I had to add an ORBIT_UNREACHABLE to the end of both functions. I think this isn't a problem because execution genuinely shouldn't reach that point and if it does then something wrong has happened and we need to give an error either way. Basically the ORBIT_UNREACHABLE at the end of the functions are just to shut clang7 and gcc10 compilers.

Reply:
Also, It seems that best practices for switch statements on protobuf enums are a bit muddled. See here: 

https://g3doc.corp.google.com/devtools/library_club/g3doc/totw/147.md?cl=head

https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#enum

Reply:
In my opinion we should keep the default case.

The problem is that protobuf's compatbility guarantees explcitly allow the field to have a different value that have no name in the current proto definition. Ignoring that fact doesn't solve the problem.

We have to decide what to do if we actually got a value that we don't know about. And I think mapping all these values to `kNotApplicable` makes sense.

The compilation error thing also doesn't work for protobuf enums, only for C/C++ enums.